### PR TITLE
add error enum to `turborepo-dirs`

### PR DESCRIPTION
**Pull request by Zertsov**
### Description
Add possibility for errors in `turborepo-dirs`.

### CODE REVIEW

**Comment by Zertsov** [comment]
* `main`
  * **#7277** <a href="https://app.graphite.dev/github/pr/vercel/turbo/7277?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Zertsov and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/turborepo-dirs/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ license = "MPL-2.0"
 
 [dependencies]
 dirs-next = "2.0.0"
+thiserror.workspace = true
```
Comment (mehulkar) [inline crates/turborepo-dirs/Cargo.toml:11]: guessing yes, but is the same as below?

```suggestion
thiserror = { workspace = true }
```

**Review by mehulkar — APPROVED** [review]
LGTM, but it would probably make sense to use it somewhere at the same time as introducing it? In case it changes as you actually start using it. Not blocking at all :ship: away

File: crates/turborepo-dirs/Cargo.toml (hunk 1 above)
Comment (Zertsov) [reply crates/turborepo-dirs/Cargo.toml:11]: Yeah they're the same, I think cargo defaults to this version if you don't specify things like `features`. I'm sure there's some configuration I can do somewhere to fix this but I haven't taken the time to deal with it yet  😁

**Comment by Zertsov** [comment]
I plan on adding a PR down this stack that uses it, but didn't wanna break a bunch of stuff since this is a change I wanna make in a couple days or so

**Pull request merged**